src/lightning_app/core/app.py

```diff
@@ -596,3 +600,62 @@ def _on_run_end(self):
         if os.getenv("LIGHTNING_DEBUG") == "2":
             del os.environ["LIGHTNING_DEBUG"]
             _console.setLevel(logging.INFO)
+
+    @staticmethod
+    def _extract_vars_from_component_name(component_name: str, state):
+        child = state
+        for child_name in component_name.split(".")[1:]:
+            if child_name in child["flows"]:
+                child = child["flows"][child_name]
+            elif "structures" in child and child_name in child["structures"]:
+                child = child["structures"][child_name]
+            elif child_name in child["works"]:
+                child = child["works"][child_name]
+            else:
+                return None
+
+        return child["vars"]
+
+    def _send_flow_to_work_deltas(self, state) -> None:
+        from lightning_app import LightningWork
+
+        if not self.flow_to_work_delta_queues:
+            return
+
+        for w in self.works:
+            if not w.has_started:
+                continue
```

Comment:
> in theory we should store the base state to diff against the last sent state to the specific work.
The last state should capture the latest state of the work.

> although I guess since in this case the work hasn’t started, then we’ll be sending out the full state when we call run, so the work will catchup.
Yes, there is no reason to send the state if the work run method hasn't been called.

> BTW, the same applies when there’s a failure to communicate of some kind.
Yes, but this is a different topic ;) This PR shouldn't be concerned with deltas lost through the queues. However, in practice I have never experienced it.